src/catalog.c

```diff
@@ -630,14 +630,22 @@ ts_catalog_restore_user(CatalogSecurityContext *sec_ctx)
  * Insert a new row into a catalog table.
  */
 void
-ts_catalog_insert(Relation rel, HeapTuple tuple)
+ts_catalog_insert_only(Relation rel, HeapTuple tuple)
 {
 	CatalogTupleInsert(rel, tuple);
 	ts_catalog_invalidate_cache(RelationGetRelid(rel), CMD_INSERT);
 	/* Make changes visible */
 	CommandCounterIncrement();
 }
 
+void
+ts_catalog_insert(Relation rel, HeapTuple tuple)
+{
+	ts_catalog_insert_only(rel, tuple);
+	/* Make changes visible */
+	CommandCounterIncrement();
```

Comment:
There is already a `CommandCounterIncrement` last in `ts_catalog_insert_only`. Not sure why you need a second one here.